src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -951,6 +951,27 @@ const WorkflowView = _.flow(
         _.update('outputs', this.isSingle() ? () => ({}) : _.mapValues(_.trim))
       )
 
+      // Dockstore users who target floating tags can change their WDL via Github without explicitly selecting a new version in Terra.
+      // If the change removes or renames inputs or outputs, the `modifiedConfig` retrieved from the DB will include stale keys that
+      // are hidden from the UI and cause the workflow to fail at launch with "Validation errors: Extra inputs".
+      //
+      // The IO set `modifiedInputsOutputs` is always current because we regenerate it from the WDL each time; use it to filter out
+      // any extraneous IO keys before saving.
+      //
+      // https://broadworkbench.atlassian.net/browse/WA-291
+      const validInputs = _.map('name', modifiedInputsOutputs['inputs'])
+      const validOutputs = _.map('name', modifiedInputsOutputs['outputs'])
+
+      const filterIO = (ioMap, allowedKeys) => {
+        return _.fromPairs(_.filter(pair => {
+          return _.includes(pair[0], allowedKeys)
+        }, _.toPairs(ioMap)))
+      }
+
+      modifiedConfig['inputs'] = filterIO(modifiedConfig['inputs'], validInputs)
```

Comment:
I think `_.pick` (https://lodash.com/docs/4.17.15#pick) will work here:

```
modifiedConfig['inputs'] = _.pick(validInputs, modifiedConfig['inputs'])
```

The only thing to be careful about is that `_.pick` takes "paths", which are a convenience for picking properties out of nested structures. I don't remember if inputs/outputs might contain periods in their names. Even if so, I suspect the inputs/outputs objects are flat objects so it should just work. If there's any concern, though, you could use `_.pickBy`:

```
modifiedConfig['inputs'] = _.pickBy((value, key) => _.includes(key, validInputs), modifiedConfig['inputs'])
```

Reply:
Agreed, and to go one step further: If you take a look at `loadNewMethodConfig` (called when changing versions), we perform this operation as part of the pipeline of changes to the `modifiedConfig` state. Since we want to do the same thing in both places, we could extract that out as a helper, e.g. something like:
```js
const filterConfigIO = ({ inputs, outputs }) => {
  return _.flow(
    _.update('inputs', _.pick(_.map('name', inputs))),
    _.update('outputs', _.pick(_.map('name', outputs)))
  )
}
```